### PR TITLE
Add flag for approximate predecessor generalization and "fix if intersects" support

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

File: engines/ic3base.cpp [hunk 1]
```diff
@@ -711,6 +712,46 @@ bool IC3Base::propagate(size_t i)
...
+    orig_pred_children = pred.children;
+    assert(orig_pred_children.size());
+  }
+
+  predecessor_generalization(i, c, pred);
+
+  if (approx_pregen_ && i >= 2) {
+    TermVec dropped;
+    assert(orig_pred_children.size());
+    TermVec pred_children = pred.children;
```
Comment (ahmed-irfan) [inline engines/ic3base.cpp:732]:
```suggestion
    const TermVec & pred_children = pred.children;
```

**Review by ahmed-irfan — APPROVED** [review]
LGTM

**Comment by makaimann** [comment]
Running jobs on the cluster for ic3sa just FYI.

**Comment by makaimann** [comment]
Screenshot from 2021-01-30 09-57-05

Looks fine, going to merge this also.

**Pull request merged**